### PR TITLE
Class Comp -> FC -- ToggleButton

### PR DESCRIPTION
**Pull request by rtragle**
This PR refactors the ToggleButton to a FC while cleaning up the code some

ToggleButton FC refactor-2020-11-18_08 43 12

https://app.clickup.com/t/g6vvcr

### CODE REVIEW

File: src/components/ToggleButton/ToggleButton.js [hunk 1]
```diff
@@ -1,134 +1,164 @@
...
+      <input type="checkbox" checked={value} onChange={onClick} name={title} />
+      <span className={classNames(className, 'slider slider--round')}></span>
+    </label>
+  );
+}
 
-  constructor(props) {
-    super(props);
+Switch.propTypes = {
+  className: PropTypes.string,
```
Comment (rtragle) [inline src/components/ToggleButton/ToggleButton.js:23]: There's a lot of repeated propTypes. I was thinking of moving all of them from the ToggleButton into a `const propTypes` and `const defaultProps` at the top.

What's your opinion @ayan4m1

Comment (rtragle) [reply src/components/ToggleButton/ToggleButton.js:23]: Actually, I kind of like this approach better:
```
IconOnly.propTypes = Switch.propTypes = GridList.propTypes = ButtonElement.propTypes = ToggleButton.propTypes = {
  className: PropTypes.string,
  buttonProps: PropTypes.object,
  iconProps: PropTypes.object,
  title: PropTypes.string.isRequired,
  value: PropTypes.bool,
  iconOnly: PropTypes.bool,
  variant: PropTypes.oneOf(toggleButtonProps.variants),
  onClick: PropTypes.func
};

IconOnly.defaultProps = Switch.defaultProps = GridList.defaultProps = ButtonElement.defaultProps = ToggleButton.defaultProps = {
  buttonProps: {},
  iconProps: {},
  value: false,
  variant: 'check'
};
```

Comment (rtragle) [reply src/components/ToggleButton/ToggleButton.js]: made the change in a24b31bdcb5545ebaf624f97fa752ff48086cbf2

File: src/components/ToggleButton/ToggleButton.js [hunk 2]
```diff
@@ -1,134 +1,109 @@
...
   }
+}
 
-  render() {
-    const { iconOnly, variant } = this.props;
-
-    if (iconOnly) {
-      return this.iconOnly;
-    }
+IconOnly.propTypes = Switch.propTypes = GridList.propTypes = ButtonElement.propTypes = ToggleButton.propTypes = {
```
Comment (rtragle) [inline src/components/ToggleButton/ToggleButton.js:93]:
```[IconOnly, Switch, GridList, ButtonElement, ToggleButton].map(
  (Comp) =>
    (Comp.propTypes = {
      className: PropTypes.string,
      buttonProps: PropTypes.object,
      iconProps: PropTypes.object,
      title: PropTypes.string.isRequired,
      value: PropTypes.bool,
      iconOnly: PropTypes.bool,
      variant: PropTypes.oneOf(toggleButtonProps.variants),
      onClick: PropTypes.func
    })
);```

I investigated doing the above to remove the redundant `.propTypes =`, but it doesn't work properly with eslint

Comment (ayan4m1) [reply src/components/ToggleButton/ToggleButton.js:93]: Oh, just declare the prop types as a constant and then assign the constant to each component's propTypes

```
const buttonProps = {
  className: PropTypes.string,
   ...
};

Icon.propTypes = buttonProps;
GridList.propTypes = buttonProps;
...
```

**Review by ayan4m1 — APPROVED** [review]
Looks good!

**Pull request merged**